swiftshield-Sources/AutomaticSwiftShield.swift

```diff
@@ -164,6 +164,21 @@ extension AutomaticSwiftShield {
                 guard obfuscationData.usrDict.contains(usr) else {
                     return
                 }
+                
+                if type == .enum && name.hasSuffix("CodingKeys") {
+                    obfuscationData.excludedEnums.insert(usr)
+                    return
+                }
+                
+                if type == .enumelement {
+                    for exclusion in obfuscationData.excludedEnums {
+                        // Enum element belongs to excluded enum
+                        if usr.hasPrefix(exclusion) {
+                            return 
+                        }
```

Comment:
Iterating every single excluded enum sounds unnecessary, since we're checking for a prefix can't we just remove the suffix and check if the remainder exists in the set?

Reply:
Do you know the mangled name rules for Swift. 
For the below example of enum element "firstName", I need to make sure I can remove the suffix "9firstNameyA2FmF" from "s:14SwiftShieldSDK6PersonC17_personCodingKeys33_7242533BA8890D95B0EE7BCC963FFFCBLLO9firstNameyA2FmF" correctly, but from what I know, the mangled name rules keeping change for different Swift version. 
> Found declaration of _personCodingKeys (s:14SwiftShieldSDK6PersonC17_personCodingKeys33_7242533BA8890D95B0EE7BCC963FFFCBLLO)
> Found declaration of firstName (s:14SwiftShieldSDK6PersonC17_personCodingKeys33_7242533BA8890D95B0EE7BCC963FFFCBLLO9firstNameyA2FmF)
> Found declaration of lastName (s:14SwiftShieldSDK6PersonC17_personCodingKeys33_7242533BA8890D95B0EE7BCC963FFFCBLLO8lastNameyA2FmF)

Reply:
I was hoping that the number that comes before the name of the element was going to be constant... In that case, you could strip everything after the name of the enumelement

Let's try stripping everything after the last underline... I suppose that's harder to change but yeah, we might have problems with other Swift versions indeed

Reply:
I just commit a new change based on your idea.